src/sentry/grouping/parameterization.py

```diff
@@ -0,0 +1,368 @@
+            |
+            # RFC3339, RFC3339Nano
+            (\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z?([+-]?\d{2}:\d{2})?)
+            |
+            # Datetime:
+            (\d{4}-?[01]\d-?[0-3]\d\s[0-2]\d:[0-5]\d:[0-5]\d)(\.\d+)?
+            |
+            # Kitchen
+            (\d{1,2}:\d{2}(:\d{2})?(?: [aApP][Mm])?)
+            |
+            # Date
+            (\d{4}-[01]\d-[0-3]\d)
+            |
+            # Time
+            ([0-2]\d:[0-5]\d:[0-5]\d)
+            |
+            # Old Date Formats, TODO: possibly safe to remove?
+            (
+                (\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z))|
+                (\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))|
+                (\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))
+            ) |
+            (
+                \b(?:(Sun|Mon|Tue|Wed|Thu|Fri|Sat)\s+)?
+                (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\s+
+                ([\d]{1,2})\s+
+                ([\d]{2}:[\d]{2}:[\d]{2})\s+
+                [\d]{4}
+            ) |
+            (
+                \b(?:(Sun|Mon|Tue|Wed|Thu|Fri|Sat),\s+)?
+                (0[1-9]|[1-2]?[\d]|3[01])\s+
+                (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\s+
+                (19[\d]{2}|[2-9][\d]{3})\s+
+                (2[0-3]|[0-1][\d]):([0-5][\d])
+                (?::(60|[0-5][\d]))?\s+
+                ([-\+][\d]{2}[0-5][\d]|(?:UT|GMT|(?:E|C|M|P)(?:ST|DT)|[A-IK-Z]))
+            ) |
+            (datetime.datetime\(.*?\))
+        """,
+    ),
+    ParameterizationRegex(name="duration", raw_pattern=r"""\b(\d+ms) | (\d(\.\d+)?s)\b"""),
+    ParameterizationRegex(name="hex", raw_pattern=r"""\b0[xX][0-9a-fA-F]+\b"""),
+    ParameterizationRegex(name="float", raw_pattern=r"""-\d+\.\d+\b | \b\d+\.\d+\b"""),
+    ParameterizationRegex(name="int", raw_pattern=r"""-\d+\b | \b\d+\b"""),
+    ParameterizationRegex(
+        name="quoted_str",
+        raw_pattern=r"""# Using `=`lookbehind which guarantees we'll only match the value half of key-value pairs,
+            # rather than all quoted strings
+            '([^']+)' | "([^"]+)"
+        """,
+        lookbehind="=",
+    ),
+    ParameterizationRegex(
+        name="bool",
+        raw_pattern=r"""# Using `=`lookbehind which guarantees we'll only match the value half of key-value pairs,
+            # rather than all instances of the words 'true' and 'false'.
+            True |
+            true |
+            False |
+            false
+        """,
+        lookbehind="=",
+    ),
+]
+
+
+DEFAULT_PARAMETERIZATION_REGEXES_MAP = {r.name: r.pattern for r in DEFAULT_PARAMETERIZATION_REGEXES}
+
+
+@dataclasses.dataclass
+class ParameterizationCallable:
+    """
+    Represents a callable that can be used to modify a string, which can give
+    us more felxibility than just using regex.
```

Comment:
```suggestion
    us more flexibility than just using regex.
```

Could you please add an example?